### PR TITLE
Add must_use to InsertResult

### LINKED ISSUE #385: Unused `InsertResult` does not generate a compiler warning.
I've found code that wrongly assumes that after an insert has been called for an Entity that calling `storage.get(entity).unwrap()` is not going to panic.  This may panic because the Entity could have been dead for the insert.  Is it possible for us to generate a compiler warning for unused `InsertResult` similar to how rustc generates a compiler warning for unused `Result`?

### PR DESCRIPTION
**Pull request by Xaeroxe**
Fixes #385

---
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/386)

### CODE REVIEW

File: src/world/mod.rs [hunk 1]
```diff
@@ -85,7 +85,10 @@ impl<'a> EntityBuilder<'a> {
     pub fn with<T: Component>(self, c: T) -> Self {
         {
             let mut storage = self.world.write_storage();
-            storage.insert(self.entity, c);
+            if let EntityIsDead(_) = storage.insert(self.entity, c) {
+                // This is technically possible, however quite unlikely.
```
Comment (torkleyy) [inline src/world/mod.rs]: I don't think it is, try yourself. It should not fail before the deletion is maintained.

File: src/storage/mod.rs [hunk 2]
```diff
@@ -109,6 +109,7 @@ where
 pub unsafe trait DistinctStorage {}
 
 /// The status of an `insert()`ion into a storage.
+#[must_use]
```
Comment (torkleyy) [inline src/storage/mod.rs]: Actually, couldn't this simply be a `Result<Option<T>>`?

Comment (torkleyy) [reply src/storage/mod.rs]: I think I wanted to do this before but decided against it because of breakage.

Comment (torkleyy) [reply src/storage/mod.rs]: My suggestion: Use the error from `error.rs` (create one if there is no such variant yet) and make this simply a type def.

File: src/world/mod.rs (hunk 1 above)
Comment (Xaeroxe) [reply src/world/mod.rs]: Oh hm, I forgot the EntityBuilder has a lifetime in it.

File: src/storage/mod.rs (hunk 2 above)
Comment (Xaeroxe) [reply src/storage/mod.rs]: Can you clarify what that type def would look like?  I'm a bit confused.

Comment (torkleyy) [reply src/storage/mod.rs]: Sorry, I shouldn't be comenting here given my limited time.

```
pub type InsertResult<T> = Result<Option<T>, MyError>;
```

**Comment by Xaeroxe** [comment]
@torkleyy I've changed `InsertResult` for the typedef, and formatted the codebase.

File: src/error.rs [hunk 3]
```diff
@@ -60,7 +60,8 @@ pub enum Error {
     /// Wrong generation error.
     WrongGeneration(WrongGeneration),
 
-    #[doc(hidden)] __NonExhaustive,
+    #[doc(hidden)]
```
Comment (torkleyy) [inline src/error.rs]: :tada:

**Comment by Xaeroxe** [comment]
@torkleyy done!

File: src/storage/tests.rs [hunk 4]
```diff
@@ -107,7 +119,7 @@ mod map_test {
         let mut w = World::new();
         let mut c = create(&mut w);
 
-        c.insert(ent(1 << 25), Comp(7));
+        let _ = c.insert(ent(1 << 25), Comp(7));
```
Comment (Xaeroxe) [inline src/storage/tests.rs]: I left the `let _ =` here because the panic we're trying to catch shouldn't come from the failed insert.

**Pull request merged**